Code/Framework/AzToolsFramework/AzToolsFramework/ViewportUi/ButtonGroup.cpp

```diff
@@ -31,11 +31,22 @@ namespace AzToolsFramework::ViewportUi::Internal
     {
         if (auto buttonEntry = m_buttons.find(buttonId); buttonEntry != m_buttons.end())
         {
-            for (auto& button : m_buttons)
-            {
-                button.second->m_state = Button::State::Deselected;
-            }
+            ClearHighlightedButton();
```

Comment:
Maybe there is an opportunity to improve in this function: first it could check if the current `m_highlightedButtonId` is already `buttonId`, if it is it doesn't need to do anything because it's already highlighted.

````
void ButtonGroup::SetHighlightedButton(ButtonId buttonId)
{
    if (m_highlightedButtonId == buttonId)
    {
        return;
    }
    ...
````